packages/uploadthing/src/sdk/utils.ts

```diff
@@ -61,9 +66,9 @@ export const uploadFilesInternal = async (
         }[];
       }
     | { error: string };
-
-  if (!res.ok || "error" in json) {
-    throw UploadThingError.fromResponse(res);
+  
+  if ("error" in json) {
+    throw UploadThingError.fromResponse(clonedRes);
```

Comment:
@markflorkowski combined with LOC 58 this should work now, although it might not be the prettiest way to handle this.

Reply:
Cloning the response is fine with me. Especially with the comment for why it is done.

Reply:
alternatively, could call .json() only `if (instanceof responseOrBody === Response)` in from Response, or add a similar method that skips the first line.

```ts
// error.ts
  public static async fromJsonBody(json: Json) {
    let message: string | undefined = undefined;
    if (json !== null && typeof json === "object" && !Array.isArray(json)) {
      if (typeof json.message === "string") {
        message = json.message;
      } else if (typeof json.error === "string") {
        message = json.error;
      }
    }
    return new UploadThingError({
      message,
      code: getErrorTypeFromStatusCode(response.status),
      cause: response,
      data: json,
    });
  }
```

Reply:
I don't think such a method provides much value, I think it's fine to just use the normal constructor if you've already parsed the body.